ddtrace/sampler.py

```diff
@@ -92,3 +103,251 @@ def set_sample_rate_by_service(self, rate_by_service):
 
 # Default key for service with no specific rate
 RateByServiceSampler._default_key = RateByServiceSampler._key()
+
+
+class DatadogSampler(BaseSampler):
+    """
+    """
+    __slots__ = ('rules', 'rate_limit')
+
+    DEFAULT_RATE_LIMIT = 100
+    NO_RATE_LIMIT = -1
+
+    def __init__(self, rules=None, rate_limit=None):
+        """
+        Constructor for DatadogSampler sampler
+
+        :param rules: List of :class:`SamplingRule` rules to apply to the root span of every trace, default no rules
+        :type rules: :obj:`list` of :class:`SamplingRule`
+        :param rate_limit: Global rate limit (traces per second) to apply to all traces regardless of the rules
+            applied to them, default 100 traces per second
+        :type rate_limit: :obj:`int`
+        """
+        # Ensure rules is a list
+        if not rules:
+            rules = []
+
+        # Validate that the rules is a list of SampleRules
+        for rule in rules:
+            if not isinstance(rule, SamplingRule):
```

Comment:
Duck typing is sad 🦆 😢

Reply:
Yeah, the reason for this is because I want this sampler to fail on `init`/application start rather than at runtime, since it won't get called (and we won't figure out if we have an incompatible rule) until we start the first trace. #protective